website/docs/icons/library/index.js

```diff
@@ -45,17 +52,45 @@ export default class Index extends Component {
             i.iconName === exactIconName && i.size === this.selectedIconSize
           );
         });
-        return icon ? [icon] : [];
+
+        if (icon) {
+          filteredIcons.push(icon);
+        }
       } else {
-        return this.allIcons.filter(
+        filteredIcons = this.allIcons.filter(
           (i) =>
             i.size === this.selectedIconSize &&
             i.searchable.indexOf(this.searchQuery) !== -1
         );
       }
     } else {
-      return this.allIcons.filter((i) => i.size === this.selectedIconSize);
+      filteredIcons = this.allIcons.filter(
+        (i) => i.size === this.selectedIconSize
+      );
+    }
+
+    // alphabetize the filtered icons
+    filteredIcons.sort((a, b) => {
+      return a.iconName.localeCompare(b.iconName);
+    });
+
+    // Group all filtered icons by category
+    filteredIcons.forEach((icon) => {
+      const category = icon.category;
+      if (!filteredGroupedIcons[category]) {
+        filteredGroupedIcons[category] = [];
+      }
+      filteredGroupedIcons[category].push(icon);
+    });
+
+    // alphabetize the grouped icon categories
+    const alphabetizedCategories = Object.keys(filteredGroupedIcons).sort();
+    const alphabetizedGroupedIcons = {};
+
+    for (const category of alphabetizedCategories) {
+      alphabetizedGroupedIcons[category] = filteredGroupedIcons[category];
     }
+    return alphabetizedGroupedIcons;
   }
```

Comment:
@didoo I added alphabetization to the icon search results

Reply:
A thought: could it be that if you alphabetize the icons (lines 72-75) by `category name` **and** `icon name`, then when you group them they're already alphabetized by categories, and within each category by name? in this way you don't have to do a second sorting for the groups and create a new object. It's worth a try?

Reply:
Refactored and pushed up, thanks.